python/pyspark/sql/dataframe.py

```diff
@@ -3795,8 +3868,11 @@ def dropDuplicates(self, subset: Optional[List[str]] = None) -> "DataFrame":
         |Alice|  5|    80|
         +-----+---+------+
         """
-        if subset is not None and (not isinstance(subset, Iterable) or isinstance(subset, str)):
-            raise TypeError("Parameter 'subset' must be a list of columns")
+        if subset is not None and not isinstance(subset, (list, tuple)):
```

Comment:
So, previously it was checking that it must be Iterable and not String, now it fails only when it is not List or Tuple. Hence means it's Ok to pass String and Tupple now?

Reply:
Only `list` or `tuple` can be passed this condition, so I believe `str` is not okay to pass. I think it's not a behavior change.

Reply:
You're right about String. What about Tuple then? It wasn't in the condition before and nor function signature mentions it.

Reply:
The previous behavior only allows `Iterable` (the intention of this is to allow `list` and `tuple` type) for `subset`, but disallows the `str`, because `str` in Python is also treated as an `Iterable` since `str` subclass of `Iterable`.

And that is not so clear to understand, so I refine this condition more clearly here.

Reply:
Got it. Just reverted the condition back.

Reply:
No more comments from my side. LGTM

Reply:
Thanks for the comment, @khalidmammadov :-)